lib/internal/modules/package_json_reader.js

```diff
@@ -123,10 +128,19 @@ function getNearestParentPackageJSON(checkPath) {
   return { data, path };
 }
 
+/**
+ * Find the nearest package.json
+ * @param {URL['pathname']} origin Where to start searching.
+ * @returns {URL['pathname']} The fully resolved location of the package.json file.
```

Comment:
Question: Do we need to also make sure this returns a `UNC` path on Windows? UNC path (the result of `toNamespacedPath()` is required if the path is extremely long. windows weirdly throws an error if long path feature is not enabled)

Reply:
```suggestion
 * @returns {URL['pathname'] | undefined} The fully resolved location of the package.json file.
```

Reply:
Regarding your question: i have no idea. It sounds like you answered it yourself though (and the answer is "yes").